### PR TITLE
[#1767] feat(netty): The client-side supports choosing Netty's ByteBufAllocator

### PR DESCRIPTION
**Pull request by rickyma**
### What changes were proposed in this pull request?

The client-side supports choosing Netty's ByteBufAllocator.

### Why are the changes needed?

For https://github.com/apache/incubator-uniffle/issues/1767.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### CODE REVIEW

File: common/src/main/java/org/apache/uniffle/common/config/RssClientConf.java [hunk 1]
```diff
@@ -132,13 +132,20 @@ public class RssClientConf {
           .defaultValue(0)
           .withDescription("Number of threads used in the client thread pool.");
 
-  public static final ConfigOption<Boolean> NETWORK_CLIENT_PREFER_DIRECT_BUFS =
+  public static final ConfigOption<Boolean> NETTY_CLIENT_PREFER_DIRECT_BUFS =
```
Comment (rickyma) [inline common/src/main/java/org/apache/uniffle/common/config/RssClientConf.java:135]: NETWORK_CLIENT_PREFER_DIRECT_BUFS
->
NETTY_CLIENT_PREFER_DIRECT_BUFS

This is more suitable.

File: common/src/main/java/org/apache/uniffle/common/config/RssClientConf.java [hunk 2]
```diff
@@ -132,13 +132,20 @@ public class RssClientConf {
...
       ConfigOptions.key("rss.client.netty.client.prefer.direct.bufs")
           .booleanType()
           .defaultValue(true)
           .withDescription(
               "If true, we will prefer allocating off-heap byte buffers within Netty.");
 
+  public static final ConfigOption<Boolean> NETTY_CLIENT_POOLED_ALLOCATOR_ENABLED =
+      ConfigOptions.key("rss.client.netty.client.pooled.allocator.enabled")
+          .booleanType()
+          .defaultValue(true)
```
Comment (rickyma) [inline common/src/main/java/org/apache/uniffle/common/config/RssClientConf.java:145]: The default value is true, so we don't change anything.

**Comment by rickyma** [comment]
@zuston @jerqi PTAL. Just add an option. No user-facing changes.

**Review by zuston — APPROVED** [review]
lgtm

**Pull request merged**